34.md

```diff
@@ -0,0 +1,84 @@
+NIP-34
+======
+
+`git` stuff
+-----------
+
+`draft` `optional`
+
+This NIP defines all the ways code collaboration using and adjacent to [`git`](https://git-scm.com/) can be done using Nostr.
+
+## Repository announcements
+
+Git repositories are hosted in Git-enabled servers, but their existence can be announced using Nostr events, as well as their willingness to receive patches, bug reports and comments in general.
+
+```jsonc
+{
+  "kind": 30617,
+  "content": "",
+  "tags": [
+    ["d", "<repo-id>"],
+    ["name", "<human-readable project name>"],
+    ["description", "brief human-readable project description>"],
+    ["head", "<sha1-hex>"], // the current repository commit head, as hex
+    ["web", "<url for browsing>"], // a webpage url, if the git server being used provides such a thing
+    ["clone", "<url for git-cloning>"], // a url to be given to `git clone` so anyone can clone it
+    ["patches", "<relay-url>"], // this signals that patches are welcome and will be searched for in this relay
+    ["issues", "<relay-url>"] // this signals that bug reports, questions, comments and so on are welcome
+  ]
+}
+```
+
+The tags `web`, `clone`, `patches` and `issues` can be specified multiple times. Except `d`, all tags are optional.
+
+## Patches
```

Comment:
how do you send a patch series?

Reply:
Isn't it simpler to send a pull request?

Reply:
I don't know? How do people normally do it in `git send-email`? I know you can send multiple patches in the same email and you could do the same in the Nostr event. Then when applying with `git am -i <patch>` it works fine.

Or you could send multiple events? I'm not sure what is the best flow.

Reply:
git send-email sends multiple emails when you give it multiple patches. it creates a proper email thread so that you can reply to each patch individually. the nostr equivalent would be building a nip-10 thread of patches.

Reply:
> Isn't it simpler to send a pull request?

this implies you have write access to the repo in question, or that you have a hosted fork somewhere. it is much easier to just send patches as it requires no hosting.

Reply:
What about a pull request event which patches tag? later updates based on feedback or rebases could also be associated with the same pull request.